kvx/src/implementations/disk.rs

```diff
@@ -93,7 +93,22 @@ impl WriteStore for Disk {
             fs::create_dir_all(dir)?;
         }
 
-        fs::write(path, format!("{:#}", value).as_bytes())?;
+        if path.exists() {
+            let tmp_file = PathBuf::from(format!("{}.swap", path.to_string_lossy()));
```

Comment:
Not sure that is safe.

Routinator creates a dedicated tmp directory under the base working directory and uses `tempfile::NamedTempFile`.

Reply:
It is _probably_ the same file system since you create all the directories under the base working directory yourself? I mean, if people mount stuff half-way down your tree, it’s on them?

Reply:
@partim could you have another look? Hopefully improved now.